components/search_results_header/search_results_header.jsx

```diff
@@ -51,16 +67,38 @@ export default class SearchResultsHeader extends React.Component {
             <div className='sidebar--right__header'>
                 <span className='sidebar--right__title'>{this.props.children}</span>
                 <div className='pull-right'>
-                    <button
-                        type='button'
-                        className='sidebar--right__expand btn-icon'
-                        aria-label='Expand'
-                        onClick={this.props.actions.toggleRhsExpanded}
+                    {
+                        this.props.icons.map(
+                            (icon) => {
+                                const id = generateId();
+                                return (
+                                    <OverlayTrigger
+                                        key={id}
+                                        delayShow={Constants.OVERLAY_TIME_DELAY}
+                                        placement='top'
+                                        overlay={<Tooltip id={id}>{icon.tooltip}</Tooltip>}
+                                    >
+                                        <button
+                                            id={id}
+                                            type='button'
+                                            className='sidebar--right__expand btn-icon'
+                                            onClick={() => this.handleIconClick(icon.action)}
```

Comment:
If the plugin authors set this action to dispatch like we do here: 
https://github.com/mattermost/mattermost-plugin-demo/blob/3ed574f17f5c67ff67843a1cef391adcf40f154f/webapp/src/plugin.jsx#L68

then here it can just be 
```suggestion
                                            onClick={() => this.icon.action()}
```
like they do for the icon in the channel header:

https://github.com/mattermost/mattermost-webapp/blob/c46b47e7e0f81c49226b9e5795a9972b1e4b43a4/plugins/channel_header_plug/channel_header_plug.jsx#L127 without the need to do a generic `mapDispatchToProps` for the action. 

I think we should follow this approach to be consistent on how the webapp api is receiving actions in the current registry functions.

Reply:
Ahh good call. Updated to pass in just a function rather than a dispatchable action.

Reply:
Updated both repos.